internal/nginx/config/servers.go

```diff
@@ -102,10 +102,11 @@ func createLocations(pathRules []dataplane.PathRule, listenerPort int) []http.Lo
 			// generate a standard location block without http_matches.
 			if len(rule.MatchRules) == 1 && isPathOnlyMatch(m) {
 				loc = http.Location{
-					Path: rule.Path,
+					Path:  rule.Path,
+					Exact: *m.Path.Type == v1beta1.PathMatchExact,
 				}
 			} else {
-				path := createPathForMatch(rule.Path, matchRuleIdx)
+				path := createPathForMatch(rule.Path, v1beta1.PathMatchType(rule.PathType), matchRuleIdx)
```

Comment:
I don't think it makes sense to convert it back to v1beta1 type here

Reply:
This is because of the `switch` statement in `formatPathType` requiring the v1beta1 type.

Reply:
does it make sense to convert it back to v1beta2 type in `formatPathType` ? :)

Reply:
Oh I see what you mean